application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/FuzzySearchSuccessTest.java

```diff
@@ -0,0 +1,163 @@
+package uk.nhs.digital.uec.api.integration.fuzzysearch;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.nhs.digital.uec.api.model.ApiSuccResponse;
+import uk.nhs.digital.uec.api.model.DosService;
+import uk.nhs.digital.uec.api.utils.MockDosServicesUtil;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@ActiveProfiles({"prod"})
+/**
+ * Test class which passes requests through the Fuzzy Search endpoint and asserts desired API
+ * behavior. Only the model layer will be mocked here.
+ */
+public class FuzzySearchSuccessTest {
+
+  @Autowired private ObjectMapper mapper;
+
+  private static String endpointUrl =
+      "http://localhost:9095/dosapi/dosservices/v0.0.1/services/byfuzzysearch";
+
+  TestRestTemplate restTemplate = new TestRestTemplate();
+  HttpHeaders headers = new HttpHeaders();
+
+  /** Sunny day scenarios. */
+
+  /**
+   * Given valid search criteria terms, ensure the API returns a 200 success code, and a list of
+   * services are returned.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void oneSearchCriteriaGiven() throws Exception {
+    HttpEntity<String> request = new HttpEntity<String>(null, headers);
+    UriComponentsBuilder uriBuilder =
+        UriComponentsBuilder.fromHttpUrl(endpointUrl).queryParam("search_criteria", "Term1");
+    ResponseEntity<String> response =
+        restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET, request, String.class);
+
+    assertTrue(response.getStatusCode() == HttpStatus.OK);
+
+    ApiSuccResponse apiSuccResponse = mapper.readValue(response.getBody(), ApiSuccResponse.class);
```

Comment:
Can we use an expressive name for this class e.g. `ApiSuccessResponse` rather than an abbreviation?

Reply:
Yes, no problem.